[새기능] 최종 합격 자동화 및 점수 순 정렬

cabbage16 opened the pull request:
## 🎫 관련 이슈
[//]: # (다음 키워드를 사용하면 해당 PR을 머지할 때 자동으로 이슈를 닫을 수 있습니다.)
[//]: # (keyword: close|closes|closed|resolve|resolves|resolved|fix|fixes|fixed)
[//]: # (예시: close #1)

close #98

<br>

## 📄 개요
[//]: # (작업 내용을 간단히 요약해서 적습니다.)
[//]: # (예시: 유저 회원가입 기능을 만들었습니다.)

>최종 합격을 최종 점수 순으로 자동으로 결정하는 기능을 만들었습니다.
>원서를 전체 조회할 때, 최종 점수 순으로 정렬하는 기능을 만들었습니다.

<br>

## 🔨 작업 내용
[//]: # (작업 내용을 자세하게 적습니다.)
[//]: # (붙임표 "-" 을 사용해서 목록을 만듭니다.)
[//]: # (예시: 유저 회원가입 API를 만들었습니다.)

- 최종 합격 자동 결정 API를 만들었습니다.
- 원서 전체 조회 시 정렬 파라미터를 추가했습니다.

<br>

## 🏁 확인 사항
[//]: # (PR을 보내기 전 다음 사항을 확인해주세요.)
[//]: # (해당 사항을 모두 이행해야 머지할 수 있습니다.)
[//]: # (- [x] 를 사용해서 완료로 표시할 수 있습니다.)

- [x] 테스트를 완료했나요?
- [x] API 문서를 작성했나요?
- [x] 코드 컨벤션을 준수했나요?
- [x] 불필요한 로그, 주석, import 등을 삭제했나요?

<br>

## 🙋🏻 덧붙일 말
[//]: # (다음 사항이 있다면 적어주세요.)
[//]: # (PR에 대한 추가 설명)
[//]: # (중점적으로 리뷰받고 싶은 부분)
[//]: # (기타 등등)
SelectFirstPassUseCaseTest처럼 만들고 싶은데 테스트 프로필에서 h2 db가 작동하지 않아서... 도움 부탁드립니다 ㅠㅠㅠ

cabbage16 committed:
feat(#98): 최종 합격 여부 결정 API 작성
- 어드민이 최종 점수 순으로 최종 합격 여부를 결정할 수 있습니다.
- 1차 합격 원서 중 최종 점수가 없는 원서가 있다면 예외가 발생합니다.

cabbage16 committed:
feat(#98): 최종 합격 여부 결정 API 작성
- 어드민이 최종 점수 순으로 최종 합격 여부를 결정할 수 있습니다.
- 1차 합격 원서 중 최종 점수가 없는 원서가 있다면 예외가 발생합니다.

cabbage16 committed:
feat(#98): 최종 합격 여부 결정 API 작성
- 어드민이 최종 점수 순으로 최종 합격 여부를 결정할 수 있습니다.
- 1차 합격 원서 중 최종 점수가 없는 원서가 있다면 예외가 발생합니다.

cabbage16 committed:
feat(#98): 최종 합격 여부 결정 API 작성
- 어드민이 최종 점수 순으로 최종 합격 여부를 결정할 수 있습니다.
- 1차 합격 원서 중 최종 점수가 없는 원서가 있다면 예외가 발생합니다.

cabbage16 committed:
refactor(#98): 사용되지 않는 import 문 제거

cabbage16 committed:
refactor(#98): 사용되지 않는 import 문 제거

cabbage16 committed:
refactor(#98): 사용되지 않는 import 문 제거

cabbage16 committed:
refactor(#98): 사용되지 않는 import 문 제거

cabbage16 committed:
refactor(#98): 예외 이름 수정
- 잘못된 이름을 수정하고 좀 더 의미에 맞도록 수정했습니다

cabbage16 committed:
refactor(#98): 예외 이름 수정
- 잘못된 이름을 수정하고 좀 더 의미에 맞도록 수정했습니다

cabbage16 committed:
perf(#98): 정원 외 전형이 다른 전형과 독립하여 경쟁하도록 수정
- 정원 외 전형은 다른 전형들과 함께 경쟁하지 않고 완전히 독립되어 경쟁해야하여서 수정했습니다.

cabbage16 committed:
perf(#98): 정원 외 전형이 다른 전형과 독립하여 경쟁하도록 수정
- 정원 외 전형은 다른 전형들과 함께 경쟁하지 않고 완전히 독립되어 경쟁해야하여서 수정했습니다.

cabbage16 committed:
test(#98): FormControllerTest 작성
- 자동으로 2차 합격 여부를 결정하는 테스트를 작성했어요.
- 2차 합격 여부를 결정할 때 최종 점수가 없는 원서가 존재하면 에러가 발생하는 테스트를 작성했어요.

cabbage16 committed:
test(#98): FormControllerTest 작성
- 자동으로 2차 합격 여부를 결정하는 테스트를 작성했어요.
- 2차 합격 여부를 결정할 때 최종 점수가 없는 원서가 존재하면 에러가 발생하는 테스트를 작성했어요.

cabbage16 committed:
docs(#98): 2차 합격 여부 자동 결정 API 문서 작성
- 2차 합격 여부를 자동으로 결정하는 API 를 추가했어요.

cabbage16 committed:
docs(#98): 2차 합격 여부 자동 결정 API 문서 작성
- 2차 합격 여부를 자동으로 결정하는 API 를 추가했어요.

cabbage16 committed:
Merge branch 'feat/#98' of https://github.com/Bamdoliro/marubase into feat/#98

cabbage16 committed:
feat(#98): 원서 전체 조회 정렬 기능 추가
- 정렬 기준을 추가해서 최종 점수 오름차순, 내림차순으로 정렬할 수 있습니다.

cabbage16 committed:
refactor(#98): 정렬 파라미터 변경
- 정렬 기준과 정렬 방식을 하나의 파라미터로 합쳤습니다.

cabbage16 committed:
test(#98): 정렬 기능 UseCaseTest 추가
- 최종 점수 내림차순 조회 테스트를 만들었어요.
- 최종 점수 오름차순 조회 테스트를 만들었어요.

cabbage16 committed:
fix(#98): 1차합격 메서드 삭제
- 테스트용으로 만든 1차합격 자동 메서드를 삭제했어요.

cabbage16 committed:
perf(#98): 최종점수순 정렬 시 최종 점수가 없는 경우 오류 발생
- 원서를 전체 조회할 때 최종 점수순으로 조회하는데 최종점수가 없는 원서가 있다면 오류를 발생하도록 했어요.

cabbage16 committed:
fix: Git Conflict 해결

cabbage16 committed:
Merge branch 'feat/#98' of https://github.com/Bamdoliro/marubase into feat/#98

cabbage16 committed:
refactor: 코드 컨벤션 준수

cabbage16 committed:
Merge branch 'develop' into feat/#98

cabbage16 committed:
fix: FormErrorProperty 콤마 해결
- 뒤에 콤마가 없어서 enum 에서 발생한 문제를 해결했어요.

gimhanul reviewed (changes requested):
로직 하나하나 뜯어보는 건 힘들 거 같고, 그냥 통합 테스트 케이스 작성해서(여러 예외 케이스 포함해서) 통과하면 머지하면 될 거 같습니다.
합격 로직은 중요하니까 꼭 통합 테스트를 할 수 있도록...!
사실 단위 테스트 / 레포지토리 테스트 / 통합 테스트 다 하면 좋을 거 같아요

cabbage16 committed:
test(#98): SelectSecondPassUseCaseTest
- 2차 합격 자동화 스프링 통합 테스트를 작성했어요.
- 스프링 부트 테스트를 위해 테스트 프로필이 있는 application.yml을 분리하여 작성했어요.

cabbage16 commented:
> 소속 하나 하나를 보내는 건 힘들 거라서, 통합 케이스 테스트 작성(여러러 컨테이너 포함해서) 통과하면 머지하면 포함되는 거 같습니다. 합격자는 중요하므로 통합 테스트를 수행할 수 있습니다...! 사실 주요 테스트 / 레포지토리 테스트 / 통합 테스트 다 그렇다는 거 같아요

통합 테스트 만들어서 총합 64명 비율에 맞게 합격 되는 것 확인했습니다! 우선 머지하고 읽으시면 코드 리뷰 부탁드립니다!
